### ISSUE TITLE
Input does not accept empty string as valid label value

### ISSUE THREAD

**Issue by Nintendot** [telus/tds-core#1121]
## Description

- Description of the defect, observations

<Input label='' ... /> will throw error:
![image](https://user-images.githubusercontent.com/20446509/59705669-d7b79780-91cc-11e9-85aa-6960d6afdb13.png)
Note: this is directly from https://tds.telus.com/components/index.html#input

- The result you were expecting
empty string should be acceptable to hide label
OR
throw a proper error

## Reproduction Steps

1. See screenshot, change label to empty string

## Reproduction Percentage (if applicable)

- 10/10 times

## Recommendation

Add accessibility documentation for core-input, and add a verbose warning when the `label` prop is neglected.

## Meta

- TDS component version: "@tds/core-input": "^3.0.1", 
- Willing to develop solution: Yes/No
- Has workaround: Yes/No 
- High impact: Yes/No 

## Screenshots

- Include any relevant screenshots

See SS above

**Comment by Nintendot** [comment]
cc: @rbrander

**Comment by theetrain** [comment]
Hello, in order to make the Input accessible, it requires a label, hence why an empty string is required. However, the error being thrown is not verbose and can be improved.

**Comment by theetrain** [comment]
Closing due to the necessity of proving a label to inputs. TDS is working on improved guidelines for form validation and client-side error handling.
